app/Repositories/OrderItemRepository.php

```diff
@@ -39,6 +40,29 @@ public function class()
      */
     public function model()
     {
-        return $this->cartItem;
+        return $this->orderItem;
+    }
+
+    public function create(array $data)
+    {
+        try {
+            return $this->model()->create($data);
+        } catch (QueryException $e) {
+            // If Integrity Constraint violation, the same item has already
+            // been added to the Order, and we should simply ignore the failure.
+
+            if ($e->getCode() !== '23000') {
```

Comment:
I don't love that this SQL error code is hard-coded and specific to MySQL/MariaDB. I suppose that we could abstract the logic further and add a method that works for every database engine. More details regarding a cross-engine approach: https://stackoverflow.com/a/26371673/1772379

As to the rationale, this unique key is intended to prevent the customer from adding a given item to the order more than once. Given that all items are digital, my position is that there is no value in ordering multiple copies of a given item (song or album).

We allowed for this in the legacy implementation, so that customers could pay artists more than requested for a given item (by increasing the quantity of a given song), and while it was used on occasion, the feature seems ripe for abuse in a high-visibility deployment.

Specifically, credit card fraudsters might be keen to order 99 copies of a song, for example, hoping to channel the payments to an artist under their control before being detected. At a minimum, preventing quantities greater than one will make that type of abuse more cumbersome for bad actors.

Reply:
So why not simply do an OR conditional here and hard code the value per supported (or possible future supported) database responses ? Is the `QueryException` here driver driven ? I assume it is where `->getCode();` will return the code on any given driver used. Which means when just need to hard code every possible code that it could be and call it a day as it will not ever likely change in the future anyways.

Reply:
Indeed; let's keep it simple. Yes, exactly, regarding `getCode()`. I'll add the codes for the other drivers that Laravel supports and call it a wrap.